gfx: Map `sans-serif` to Helvetica on Mac and DejaVu Sans on Linux.

pcwalton opened the pull request:
This matches what I believe the OS native defaults to be.

Partially addresses #9487.

r? @metajack 
cc @paulrouget

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/10937)

<!-- Reviewable:end -->

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify gfx code, but no tests are modified. Please consider adding a test!

pcwalton committed:
gfx: Map `sans-serif` to Helvetica on Mac, DejaVu Sans on Linux, and
Roboto on Android.

This matches what I believe the OS native defaults to be.

Partially addresses #9487.

metajack commented:
@bors-servo r+

I'm curious why we aren't using font-config for this mapping since that seems to be its function. But this looks good.

bors-servo commented:
:pushpin: Commit c46c59a has been approved by `metajack`

<!-- @bors-servo r=metajack c46c59a365a521ea3ba986cdd505611a59a09d56 -->

bors-servo commented:
:hourglass: Testing commit c46c59a with merge 727bf7a...

bors-servo commented:
:broken_heart: Test failed - [android](http://build.servo.org/builders/android/builds/648)

pcwalton commented:
r? @metajack 

Added Android.

metajack commented:
@bors-servo r+

---

Reviewed 3 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10937#-:-KGooUWrWhmftZgZ1LFr:1236647940)*
<!-- Sent from Reviewable.io -->

bors-servo commented:
:pushpin: Commit bcdb53e has been approved by `metajack`

<!-- @bors-servo r=metajack bcdb53e8d3353b7d3cfec88c6a552b354b4ca104 -->

bors-servo commented:
:hourglass: Testing commit bcdb53e with merge 10f043d...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/711)

highfive commented:
▶ FAIL [expected PASS] /css-text-3_dev/html/word-break-normal-bo-000.htm
      └   → /css-text-3_dev/html/word-break-normal-bo-000.htm 4f0df934a67a74b4001e44416adbd0870ca317ff
    /css-text-3_dev/html/reference/word-break-normal-bo-ref-000.htm 094fa50bd8bc5fc0817433414d3548933e9ed251
    Testing 4f0df934a67a74b4001e44416adbd0870ca317ff == 094fa50bd8bc5fc0817433414d3548933e9ed251

jdm commented:
In the reference both boxes look like the second one:
<img width="455" alt="screen shot 2016-05-03 at 11 57 58 am" src="https://cloud.githubusercontent.com/assets/27658/14989521/5166a07e-1126-11e6-8f5b-687a282252f8.png">

highfive commented:
New code was committed to pull request.

pcwalton commented:
@bors-servo: r=metajack

The test was already failing on Linux and if it was font-sensitive it probably wasn't working anyway, so let's just mark it as failing everywhere.

bors-servo commented:
:pushpin: Commit 27b2419 has been approved by `metajack`

<!-- @bors-servo r=metajack 27b2419c5a3d1db8f592669d66a4d10ed2bba010 -->

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge ee4d9ee...

bors-servo commented:
:broken_heart: Test failed - [windows](http://build.servo.org/builders/windows/builds/447)

ConnorGBrewster commented:
@bors-servo retry
 * git

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge b5889a5...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/795)

highfive commented:
▶ FAIL [expected PASS] /_mozi</span><span class="stdout">lla/css/input_selection_incremental_a.html
      └   → /_mozilla/css/input_selection_incremental_a.html 77206f4ef5cf4394f733dae85b67e229dd8e8cc6
    /_mozilla/css/input_selection_incremental_ref.html 91db21a5d4b097bc5bee206f0d4bd2c977457ff1
    Testing 77206f4ef5cf4394f733dae85b67e229dd8e8cc6 == 91db21a5d4b097bc5bee206f0d4bd2c977457ff1

jdm commented:
The non-reference text is offset a pixel or two lower than the reference.

jdm commented:
Seems like that would have to be an existing intermittent failure.

jdm commented:
@bors-servo: retry
* #11025

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge 669a869...

bors-servo commented:
:broken_heart: Test failed - [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/782)

jdm commented:
@bors-servo: retry
* infra

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge 5ce9e48...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/804)

highfive commented:
▶ FAIL [expected PASS] /_mozilla/css/input_selection_incremental_a.html
      └   → /_mozilla/css/input_selection_incremental_a.html 77206f4ef5cf4394f733dae85b67e229dd8e8cc6
    /_mozilla/css/input_selection_incremental_ref.html 91db21a5d4b097bc5bee206f0d4bd2c977457ff1
    Testing 77206f4ef5cf4394f733dae85b67e229dd8e8cc6 == 91db21a5d4b097bc5bee206f0d4bd2c977457ff1

ConnorGBrewster commented:
@bors-servo retry
 * #11025

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge 3985525...

bors-servo commented:
:broken_heart: Test failed - [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/799)

ConnorGBrewster commented:
@bors-servo retry
 * infra

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge 1fba94c...

bors-servo commented:
:broken_heart: Test failed - [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/805)

jdm commented:
@bors-servo: retry
* servo/saltfs#349

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge f023a56...

bors-servo commented:
:broken_heart: Test failed - [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/808)

ConnorGBrewster commented:
@bors-servo retry
 * servo/saltfs#349

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge d5fc08f...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/840)

highfive commented:
▶ Unexpected subtest result in /_mozilla/mozilla/iframe_contentDocument.html:
      │ FAIL [expected PASS] Iframe contentDocument test.
      │   → iframe.contentDocument.getElementById(...) is null
      │ 
      │ test_contentWindow@http://web-platform.test:8000/_mozilla/mozilla/iframe_contentDocument.html:11:19
      │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1397:20
      └ Test.prototype.step_func/&lt;@http://web-platform.test:8000/resources/testharness.js:1421:20

jdm commented:
@bors-servo: retry
* #11036

metajack commented:
Gotta keep hope alive!

bors-servo commented:
:zap: Previous build results for [android](http://build.servo.org/builders/android/builds/737), [arm32](http://build.servo.org/builders/arm32/builds/790), [arm64](http://build.servo.org/builders/arm64/builds/776), [linux-dev](http://build.servo.org/builders/linux-dev/builds/793), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/811), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/787), [windows](http://build.servo.org/builders/windows/builds/475) are reusable. Rebuilding only [linux-rel](http://build.servo.org/builders/linux-rel/builds/840), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/828)...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/830)

highfive commented:
▶ FAIL [expected PASS] /_mozilla/css/input_selection_incremental_a.html
      └   → /_mozilla/css/input_selection_incremental_a.html 77206f4ef5cf4394f733dae85b67e229dd8e8cc6
    /_mozilla/css/input_selection_incremental_ref.html 91db21a5d4b097bc5bee206f0d4bd2c977457ff1
    Testing 77206f4ef5cf4394f733dae85b67e229dd8e8cc6 == 91db21a5d4b097bc5bee206f0d4bd2c977457ff1

ConnorGBrewster commented:
@bors-servo retry
 * #11025

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge 8c1c5f5...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/831)

highfive commented:
▶ FAIL [expected PASS] /_mozilla/css/input_selection_incremental_a.html
      └   → /_mozilla/css/input_selection_incremental_a.html 77206f4ef5cf4394f733dae85b67e229dd8e8cc6
    /_mozilla/css/input_selection_incremental_ref.html 91db21a5d4b097bc5bee206f0d4bd2c977457ff1
    Testing 77206f4ef5cf4394f733dae85b67e229dd8e8cc6 == 91db21a5d4b097bc5bee206f0d4bd2c977457ff1

metajack commented:
@bors-servo retry

- #11025

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge 56c141d...

bors-servo commented:
:broken_heart: Test failed - [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/821)

ConnorGBrewster commented:
@bors-servo retry
 * servo/saltfs#349

Crossing my fingers...

bors-servo commented:
:hourglass: Testing commit 27b2419 with merge 42abc04...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/842)

highfive commented:
▶ FAIL [expected PASS] /_mozilla/css/input_selection_incremental_a.html
      └   → /_mozilla/css/input_selection_incremental_a.html 77206f4ef5cf4394f733dae85b67e229dd8e8cc6
    /_mozilla/css/input_selection_incremental_ref.html 91db21a5d4b097bc5bee206f0d4bd2c977457ff1
    Testing 77206f4ef5cf4394f733dae85b67e229dd8e8cc6 == 91db21a5d4b097bc5bee206f0d4bd2c977457ff1

jdm commented:
The input_selection_incremental tests use `font-family: sans-serif` and hasn't failed on any other PR before this. We'll need to address it before continuing to try to merge this.

metajack commented:
@pcwalton Can you investigate?

highfive commented:
New code was committed to pull request.

pcwalton commented:
The test was failing because the line-height of Helvetica is different from that of Arial. Moving the font in the reference to an attribute on an inline-block to match the `<input>` element of the test fixed the problem. Note that the test was failing in Gecko for the same reason, and with this fix the test now passes in Gecko.

pcwalton commented:
@bors-servo: r=metajack

bors-servo commented:
:pushpin: Commit 2ac9747 has been approved by `metajack`

<!-- @bors-servo r=metajack 2ac974702394c2c8e7804bd634f644b36ad9be3f -->

metajack commented:
So the test was only failing on one platform, and now it fails on all of them. Your comment seemed to imply we now pass.

<details><summary><i>Previously, bors-servo wrote…</i></summary>
> :pushpin: Commit 2ac9747 has been approved by `metajack`
> 
> <!-- @bors-servo r=metajack 2ac974702394c2c8e7804bd634f644b36ad9be3f --></details>


---

Reviewed 1 of 1 files at r3.
Review status: 4 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10937#-:-KHM2w62tMcxbevulOP2:-2139610631)*
<!-- Sent from Reviewable.io -->

bors-servo commented:
:bulb: This pull request was already approved, no need to approve it again.

- There's another pull request that is currently being tested, blocking this pull request: #11090

bors-servo commented:
:hourglass: Testing commit 2ac9747 with merge 180a981...

pcwalton commented:
Hmm? Sorry, I meant the incremental input test, not the word break one.

metajack commented:
Weird. Reviewable didn't show changes to the other test. Why does the word break test now also fail on OS X?

pcwalton commented:
It was only accidentally passing before. It's a test of Indic script word breaking rules, and we don't display Indic fonts yet.

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/807), [arm32](http://build.servo.org/builders/arm32/builds/872), [arm64](http://build.servo.org/builders/arm64/builds/851), [linux-dev](http://build.servo.org/builders/linux-dev/builds/872), [linux-rel](http://build.servo.org/builders/linux-rel/builds/921), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/895), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/872), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/914), [windows](http://build.servo.org/builders/windows/builds/570)
